python/servo/command_base.py

```diff
@@ -33,6 +35,54 @@ def cd(new_path):
         os.chdir(previous_path)
 
 
+@contextlib.contextmanager
+def setlocale(name):
+    """Context manager for changing the current locale"""
+    saved_locale = locale.setlocale(locale.LC_ALL)
+    try:
+        yield locale.setlocale(locale.LC_ALL, name)
+    finally:
+        locale.setlocale(locale.LC_ALL, saved_locale)
+
+
+def archive_deterministically(dir_to_package, dest_archive, prepend_path=None):
+    """Create a .tar.gz package in a deterministic (reproducible) manner.
+
+    See https://reproducible-builds.org/docs/archives/ for more details. """
+
+    def reset(tarinfo):
+        """Helper to reset owner/group and modification time for tar entries"""
+        tarinfo.uid = tarinfo.gid = 0
+        tarinfo.uname = tarinfo.gname = "root"
+        tarinfo.mtime = 0
+        return tarinfo
+
+    with cd(dir_to_package):
```

Comment:
I didn't realize we'd need to strip the prefixes, it does sound like `cd` is the better option here.

Using `cd` during builds adds another piece of state to keep track of, which makes it hard to interactively run commands from the middle of a script to debug/test them, since you may have changed directories much earlier; it's also easy to forget to `cd` back. Arguably, this is mostly a problem for shell scripts, and not as much of an issue for Python, both due to the easy-to-see and automatic (safe) scoping of `with cd()` and because I'm less likely to interactively run things like `with gzip.GzipFile(..):`.